### PR TITLE
Convert to using click for ce_install

### PR DESCRIPTION
**Pull request by mattgodbolt**
also tidy up a few minor naming things.

### CODE REVIEW

**Comment by mattgodbolt** [comment]
It's a big change but _mostly_ moving code about.

File: bin/lib/ce_install.py [hunk 1]
```diff
@@ -66,256 +90,324 @@ def squash_mount_check(rootfolder, subdir, context):
...
-    elif args.command == 'addcrate':
-        libyaml = LibraryYaml(args.yaml_dir)
-        libyaml.add_rust_crate(args.filter[0], args.filter[1])
-        libyaml.save()
-
-    else:
-        raise RuntimeError("Er, whoops")
+
+def main():
+    cli(prog_name='ce_install')  # pylint: disable=unexpected-keyword-arg,no-value-for-parameter
```
Comment (apmorton) [inline bin/lib/ce_install.py:410]: you can add some lines to pylintrc to make this do the correct thing: https://stackoverflow.com/a/72212721

**Review by apmorton — APPROVED** [review]
That oughta keep my account active for another few years, eh?

File: bin/lib/ce_install.py (hunk 1 above)
Comment (mattgodbolt) [reply bin/lib/ce_install.py:410]: thanks!

**Pull request merged**